compaction/compaction_manager.cc

```diff
@@ -956,6 +961,64 @@ void compaction_manager::submit(replica::table* t) {
     (void)perform_task(make_shared<regular_compaction_task>(*this, t));
 }
 
+bool compaction_manager::can_perform_regular_compaction(replica::table* t) {
+    if (!can_proceed(t) || t->is_auto_compaction_disabled_by_user()) {
+        return false;
+    }
+    auto& lk = get_compaction_state(t).lock;
+    auto could_lock = lk.try_read_lock();
+    if (!could_lock) {
+        return false;
+    }
```

Comment:
now that regular can proceed in parallel to major, perhaps we can remove this code which mimicks the serialization logic?

Reply:
agreed